### PR TITLE
chunk # #3731

### LINKED ISSUE #3622: chunk

### PR DESCRIPTION
**Pull request by Damilola-ops**
closes #3622

### CODE REVIEW

File: ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py [hunk 1]
```diff
@@ -20,6 +20,10 @@ def nonzero(input, *, out=None, as_tuple=False):
     return ret
 
 
+def chunk(input, chunks, dim=0):
+    return ivy.split(input, num_or_size_splits=chunks, dim=dim, with_remainder=True)
```
Comment (juliagsy) [inline ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py:24]: it's `axis=dim`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 2]
```diff
@@ -456,3 +456,44 @@ def test_torch_swapaxes(
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend='torch',
+        fn_tree='chunk',
+        input=np.asarray(value,dtype=input_dtype),
+        dim=dim,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:498]: `axis=dim`

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Great work so far! Just some small changes and you're almost good to go! As usual, please request my review again once you're done! Thank you!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 3]
```diff
@@ -456,3 +456,44 @@ def test_torch_swapaxes(
         axis0=axis0,
         axis1=axis1,
     )
+
+
+#chunk
+@given(
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:462]: please also add `@handle_cmd_line_args` above this line, thanks!

**Review by Damilola-ops — COMMENTED** [review]
alright, thanks so much

**Review by Damilola-ops — COMMENTED** [review]
added cmd line arguments decorator and changed axis=dm

**Comment by juliagsy** [comment]
have you pushed your change to your fork's master? cuz currently I can't see it 😅

**Comment by Damilola-ops** [comment]
> yes I have

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 4]
```diff
@@ -456,3 +456,45 @@ def test_torch_swapaxes(
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend='torch',
+        fn_tree='chunk',
+        input=np.asarray(value,dtype=input_dtype),
+        dim=dim,
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:499]: please generate data for the `chunks` parameter and add the argument here

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Thanks for the changes! One argument is left out for testing, once that's amended and tests are passing, you'll be good to go! Please request my review once you're done! Thanks!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py (hunk 4 above)
Comment (Damilola-ops) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:518]: Hi , thanks for the review , how do I generate data for the parameter ?, I'm a little lost on that

Comment (juliagsy) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:518]: through the helpers function, just like how you did for the input arrays with `helpers.dtype_and_values`, more can be found in the documentation!
https://lets-unify.ai/ivy/deep_dive/16_ivy_frontends_tests.html

**Review by juliagsy — COMMENTED** [review]
I've commented under the review comment, thanks!

**Review by juliagsy — COMMENTED** [review]
Hey there! Sorry but I still can't see your new changes in this PR 😅 If you're having issues pushing the changes here, you can open a new PR with your latest updates, refer this PR and related issue in the new PR, then close this old PR. Thank you!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py [hunk 5]
```diff
@@ -359,83 +359,47 @@ def test_torch_transpose(
...
         fw=fw,
-        frontend="torch",
-        fn_tree="swapaxes",
-        input=np.asarray(value, dtype=input_dtype),
-        axis0=axis0,
-        axis1=axis1,
-    )
+        frontend='torch',
+        fn_tree='chunk',
+        input = np.asarray(value,dtype=input_dtype),
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:401]: args should follow frontend sequence and name, which is `input, chunks, dim`

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! I noticed you've implemented the chunk in the backend. Actually, you should not implement anything in the backend, and since it is behaving as `ivy.split` so your previous approach is correct.

As for the tests, it seems you have removed the swapaxes (and possibly squeeze) test, please place them back and only make changes related to your function 😊

Thanks!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py (hunk 5 above)
Comment (Damilola-ops) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:401]: okay , just changed that , thanks so much

Comment (Damilola-ops) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py:402]: Hi , I have had issues reverting back the changes I made to swapaxes and squeeze(accidentally deleted them while resolving conflicts)  ,however  I have committed all the changes you suggested to the test functions .

**Review by juliagsy — APPROVED** [review]
Thanks for contributing! Will merge in soon!

**Pull request merged**